c-api.rst

```diff
@@ -0,0 +1,191 @@
+=======================
+Changing Python's C API
+=======================
+
+The C API is divided into three sections:
+
+1. The internal, private API, available with ``Py_BUILD_CORE`` defined.
+   Ideally declared in ``Include/internal/``. Any API named with a leading
+   underscore is also considered private.
+2. The public C API, available when ``Python.h`` is included normally.
```

Comment:
Maybe drop _"normally"_?

Reply:
By *normally* I mean *without defining macros to select the other variants*. I'll expand that in the section below.